pkg/kubectl/cmd/util/factory.go

```diff
@@ -149,6 +149,7 @@ type ClientAccessFactory interface {
 	// SuggestedPodTemplateResources returns a list of resource types that declare a pod template
 	SuggestedPodTemplateResources() []schema.GroupResource
 
+	// TODO: how is this different than PrinterForMapping?
 	// Returns a Printer for formatting objects of the given type or an error.
 	Printer(mapping *meta.RESTMapping, options printers.PrintOptions) (printers.ResourcePrinter, error)
```

Comment:
@fabianofranz maybe we could get rid of this method and use `PrinterForMapping` instead?